core/src/main/scala/com/microsoft/azure/synapse/ml/logging/SynapseMLLogging.scala

```diff
@@ -76,20 +76,34 @@ trait SynapseMLLogging extends Logging {
     logBase("constructor", None)
   }
 
-  def logFit[T](f: => T, columns: Int): T = {
-    logVerb("fit", f, columns)
+  def logFit[T](f: => T, columns: Int, logCertifiedEvent: Boolean = true,
+                certifiedEventAttributes: Map[String, String] = Map()): T = {
+    logVerb("fit", f, columns, logCertifiedEvent, certifiedEventAttributes)
   }
 
-  def logTrain[T](f: => T, columns: Int): T = {
-    logVerb("train", f, columns)
+  def logTrain[T](f: => T, columns: Int, logCertifiedEvent: Boolean = true,
+                  certifiedEventAttributes: Map[String, String] = Map()): T = {
+    logVerb("train", f, columns, logCertifiedEvent, certifiedEventAttributes)
   }
 
-  def logTransform[T](f: => T, columns: Int): T = {
-    logVerb("transform", f, columns)
+  def logTransform[T](f: => T, columns: Int, logCertifiedEvent: Boolean = true,
+                      certifiedEventAttributes: Map[String, String] = Map()): T = {
+    logVerb("transform", f, columns, logCertifiedEvent, certifiedEventAttributes)
   }
-  def logVerb[T](verb: String, f: => T, columns: Int = -1): T = {
+  def logVerb[T](verb: String, f: => T, columns: Int = -1, logCertifiedEvent: Boolean = true,
+                 certifiedEventAttributes: Map[String, String] = Map()): T = {
     logBase(verb, if(columns == -1) None else Some(columns))
     try {
+      // Begin emitting certified event.
+      if(logCertifiedEvent)
+      {
+        import com.microsoft.azure.synapse.ml.logging.Usage.FeatureSynapseML
```

Comment:
Put imports at top of code

Reply:
Sure. I put that in here so that they are only available in that small scope.